velox/functions/lib/Re2Functions.cpp

```diff
@@ -378,7 +379,28 @@ class LikeConstantPattern final : public VectorFunction {
  public:
   LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
-            RE2::Quiet) {}
+            RE2::Quiet) {
+    pattern_ = pattern;
+    fixedPattern_ = false;
+    prefixPattern_ = false;
+    suffixPattern_ = false;
+
+    if (escapeChar) {
+      return;
+    }
+    if ((fixedPattern_ =
+             !likeutils::isWildcardInPattern(pattern_) ? true : false)) {
+      return;
+    }
+    if ((prefixPattern_ =
+             likeutils::isPrefixPattern(pattern_) ? true : false)) {
+      return;
+    }
+    if ((suffixPattern_ =
```

Comment:
I wonder if it would be more efficient to detect the type of the pattern in a single pass other the 'pattern' string. It may also be convenient to use an enum for the types of patterns.

```
enum class PatternKind {
  kFixed,
  kPrefix,
  kSuffix,
  kGeneric,
}

PatternKind determinePatternKind(StringView pattern)
```